src/test/java/perosnal/spotifymatcher/service/UserServiceTest.java

```diff
@@ -0,0 +1,131 @@
+package perosnal.spotifymatcher.service;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import perosnal.spotifymatcher.model.User;
+import perosnal.spotifymatcher.repository.UserRepository;
+import perosnal.spotifymatcher.util.AuthorizedActionResult;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.lenient;
+
+@ExtendWith(MockitoExtension.class)
+class UserServiceTest {
+
+    @Mock
+    private UserRepository userRepository;
+    @Mock
+    private SpotifyApiService spotifyApiService;
+    @InjectMocks
+    private UserService userService;
+
+    @BeforeEach
+     void before() {
+        lenient().when(spotifyApiService.getIdByToken("")).thenReturn("");
+        lenient().when(userRepository.getById("")).thenReturn(new User());
+    }
+
+    @Test
+    void given_short_bio_should_return_failure() {
+        AuthorizedActionResult result = userService.setBio("", "123");
+        assertEquals(result, AuthorizedActionResult.FAILURE);
+    }
+
+    @Test
+    void given_null_bio_should_return_failure() {
+        AuthorizedActionResult result = userService.setBio("", null);
+        assertEquals(result, AuthorizedActionResult.FAILURE);
+    }
+
+    //Biography should be 20 characters or above
+    @Test
+    void given_valid_bio_should_return_success() {
+        AuthorizedActionResult result = userService.setBio("", "123456789Test123456789");
+        assertEquals(result, AuthorizedActionResult.SUCCESS);
+    }
```

Comment:
Tests could be parameterized:

```suggestion
    @ParameterizedTest(name = "given bio {0} setBio should return {1}")
    @MethodSource("givenBioToExpectedSetBioResult")
    void setBio_shouldReturnExpectedResult(final String bio, final AuthorizedActionResult expectedResult) {
        // given
        when(spotifyApiService.getIdByToken("")).thenReturn("");
        when(userRepository.getById("")).thenReturn(new User());

        // when
        final AuthorizedActionResult actualResult = userService.setBio("", bio);

        // then
        assertEquals(actualResult, expectedResult);
    }

    static Stream<Arguments> givenBioToExpectedSetBioResult() {
        return Stream.of(
                arguments("", AuthorizedActionResult.FAILURE),
                arguments("too short", AuthorizedActionResult.FAILURE),
                arguments(null, AuthorizedActionResult.FAILURE),
                arguments("123456789Test123456789", AuthorizedActionResult.SUCCESS)
        );
    }
```